server/src/stateManagement.mjs

```diff
@@ -179,38 +178,37 @@ function getUserId(userId){
   return userId;
 }
 
-// return state based on hierarchy (From lowest priority to highest) global->group->user
-function getState(userId) {
+// return state based on hierarchy (From lowest priority to highest) global->group->user if mergedState=true
+//return state of the UserId if mergedState=false,to update the state
+function getState(userId,mergedState = true) {
   userId = "" + userId;
-
-  //to get the userId from given user/appId
+//to get the userId from given user/appId
   userId = getUserId(userId);
-  if ( userId in state ) {
-    const stateCopy = JSON.parse( JSON.stringify(state) )
-    let finalState = stateCopy['global'];
-    userId = '' + userId;
-    if( userId.includes("~")){
-      let group = "~"+userId.split("~")[1];
-      if (group.includes("#")){
-        group = group.split("#")[0];
+  if(!mergedState){
+    return state[''+userId];
+  }else{
+    if ( userId in state ) {
+      const stateCopy = JSON.parse( JSON.stringify(state) )
+      let finalState = stateCopy['global'];
+      //Parsing the UserId to get group and appid
+      let parseUserId=parseUser(userId)
+      if(!(parseUserId.group == undefined && parseUserId.appId== undefined))
+      {
+        if(parseUserId.group || parseUserId.appId in stateCopy)
+        {
+          let groupOrAppState = parseUserId.appId ? stateCopy['#'+parseUserId.appId]:stateCopy['~'+parseUserId.group];
```

Comment:
Why are we dealing with app state here? Getting the user state for an appId is done in getUserId() on line 186. We should only need to reference the group here.

Reply:
As per the previous existing logic, If the group exist in the userId, we are assigning the group to the "group" variable and it it getting overridden when the appId exist in the userId.So I have rewritten the same logic, to "groupOrAppState"variable will have value as appId if it is exist in the userId, else it will have group. 

Here is the previous existing code before change:

```if( userId.includes("~")){
      let group = "~"+userId.split("~")[1];    ---//group variable is assigned to the group in userId
      if (group.includes("#")){
        group = group.split("#")[0];  ----------//group variable is assigned to the appId in userId
      }
      if (group in stateCopy){
        let groupState = stateCopy[''+group];
        logger.info(`Group State in grp of user${userId} ${JSON.stringify(groupState)}`);
        resetSequenceStateValues(finalState, groupState);
        mergeWith(finalState, groupState, mergeCustomizer);
        logger.info(`Final state in grp of user${userId} ${JSON.stringify(finalState)}`);
      }

Do you want me to change like below?
 if(!(parseUserId.group == undefined)
      {
        if(parseUserId.group in stateCopy)
        {
          let groupOrAppState = stateCopy['~'+parseUserId.group];
          resetSequenceStateValues(finalState, groupOrAppState);
          mergeWith(finalState, groupOrAppState, mergeCustomizer);
        }
    }

Reply:
Done with the changes.